elastica/rigidbody/mesh_rigid_body.py

```diff
@@ -0,0 +1,198 @@
+__doc__ = """rigid body class based on mesh"""
+
+import numpy as np
+import numba
+from elastica._linalg import _batch_cross, _batch_norm
+from elastica.utils import MaxDimension
+from elastica.rigidbody.rigid_body import RigidBodyBase
+
+
+class MeshRigidBody(RigidBodyBase):
+    def __init__(
+        self,
+        mesh,
+        center_of_mass,
+        mass_second_moment_of_inertia,
+        density,
+        volume,
+    ):
+        """
+        Mesh rigid body initializer.
+
+        Parameters
+        ----------
+        mesh
+        center_of_mass
+        mass_second_moment_of_inertia
+        density
+        volume
+        """
+        # rigid body does not have elements it only have one node. We are setting n_elems to
+        # zero for only make code to work. _bootstrap_from_data requires n_elems to be defined
+        self.n_elems = 1  # center_mass
+
+        self.density = density
+        self.volume = volume
+        self.mass = np.array([self.volume * self.density])
+        normal = np.array([1.0, 0.0, 0.0]).reshape(3, 1)
+        tangents = np.array([0.0, 0.0, 1.0]).reshape(3, 1)
+        binormal = _batch_cross(tangents, normal)
+        self.faces = mesh.faces.copy()
+        self.n_faces = mesh.faces.shape[-1]
+        self.face_centers = np.array(mesh.face_centers.copy())
+        self.face_normals = np.array(mesh.face_normals.copy())
+        self.distance_to_face_centers = _batch_norm(
+            self.face_centers - center_of_mass.reshape(3, 1)
+        )
+        self.direction_to_face_centers = (
+            self.face_centers - center_of_mass.reshape(3, 1)
+        ) / self.distance_to_face_centers
```

Comment:
Don't we need to convert this in material frame?

Reply:
The director of the mesh rigid body is always initialized as the identity (same as lab frame) so no conversion is needed.